test/HttpServer.test.ts

```diff
@@ -0,0 +1,112 @@
+import { stub, createStubInstance, SinonStubbedInstance } from 'sinon';
+import { use, assert, request, expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import chaiHttp from 'chai-http';
+import {
+    RelayServer,
+    HttpServer,
+    ServerDependencies,
+    TxStoreManager,
+    KeyManager,
+    ServerConfigParams
+} from '../src';
+import { ContractInteractor } from '@rsksmart/rif-relay-common';
+
+use(chaiAsPromised);
+use(chaiHttp);
+
+describe('HttpServer', () => {
+    const fakeServerConfig: Partial<ServerConfigParams> = {
+        url: 'http://localhost:8090'
+    };
+    const port = 8095;
+    const gasPrice = 5;
+    const method: keyof RelayServer = 'getMinGasPrice';
+    let httpServer: HttpServer;
+    let fakeRelayServer: RelayServer;
+    let fakeKeyManager: SinonStubbedInstance<KeyManager>;
+    let fakeContractInteractor: ContractInteractor;
+    let fakeStoreManager: TxStoreManager;
+    let mockDependencies: ServerDependencies;
+
+    before(() => {
+        fakeKeyManager = createStubInstance(KeyManager, {
+            getAddress: 'fakeAddress'
+        });
+        mockDependencies = {
+            managerKeyManager: fakeKeyManager,
+            workersKeyManager: fakeKeyManager,
+            contractInteractor: fakeContractInteractor,
+            txStoreManager: fakeStoreManager
+        };
+        fakeRelayServer = new RelayServer(fakeServerConfig, mockDependencies);
+        httpServer = new HttpServer(port, fakeRelayServer);
+    });
```

Comment:
I think this is meant to be a beforeEach hook, since you want to have a clean slate for each test.

Reply:
Also if you're assigning it to this mockDependencies object then there is no need to keep the `fakeKeyManager` in higher scope, since you have a reference to that very object in both `manager` and `workers` properties.

Reply:
actually non of that is used outside the hook, so you can just declare it here as well.

Reply:
updated